src/scheduling/appointments/AppointmentDetailForm.tsx

```diff
@@ -13,42 +13,42 @@ interface Props {
   appointment: Appointment
   patient?: Patient
   isEditable: boolean
-  onAppointmentChange: (appointment: Appointment) => void
+  errorMessage?: string
+  onFieldChange?: (key: string, value: string | boolean) => void
 }
 
 const AppointmentDetailForm = (props: Props) => {
-  const { onAppointmentChange, appointment, patient, isEditable } = props
+  const { onFieldChange, appointment, patient, isEditable, errorMessage } = props
   const { t } = useTranslation()
+
+  const onSelectChange = (event: React.ChangeEvent<HTMLSelectElement>, fieldName: string) =>
+    onFieldChange && onFieldChange(fieldName, event.target.value)
+
+  const onDateChange = (date: Date, fieldName: string) =>
+    onFieldChange && onFieldChange(fieldName, date.toISOString())
+
+  const onInputElementChange = (event: React.ChangeEvent<HTMLInputElement>, fieldName: string) =>
+    onFieldChange && onFieldChange(fieldName, event.target.value)
+
   return (
     <>
+      {errorMessage && <Alert color="danger" message={errorMessage} />}
       <div className="row">
         <div className="col">
           <div className="form-group">
-            {isEditable && !patient ? (
-              <>
-                <Label htmlFor="patientTypeahead" text={t('scheduling.appointment.patient')} />
-                <Typeahead
-                  id="patientTypeahead"
-                  value={appointment.patientId}
-                  placeholder={t('scheduling.appointment.patient')}
-                  onChange={(p: Patient[]) => {
-                    onAppointmentChange({ ...appointment, patientId: p[0].id })
-                  }}
-                  onSearch={async (query: string) => PatientRepository.search(query)}
-                  searchAccessor="fullName"
-                  renderMenuItemChildren={(p: Patient) => (
-                    <div>{`${p.fullName} (${p.friendlyId})`}</div>
-                  )}
-                />
-              </>
-            ) : (
-              <TextInputWithLabelFormGroup
-                name="patient"
-                label={t('scheduling.appointment.patient')}
-                value={patient?.fullName}
-                isEditable={isEditable}
-              />
-            )}
+            <Label htmlFor="patientTypeahead" text={t('scheduling.appointment.patient')} />
+            <Typeahead
+              id="patientTypeahead"
+              disabled={!isEditable || patient !== undefined}
```

Comment:
This line appears to be why the Travis build is failing, because it doesn't recognize `disabled` as a prop. I recently added `disabled` in https://github.com/HospitalRun/components/pull/290. Is Travis running it with an older version of the components library?

Reply:
Seems like it, since GitHub actions passed. @tehKapa can you check this out?